main.cpp

```diff
@@ -18,13 +18,34 @@ using namespace clang::ast_matchers;
 using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
+private:
+    Rewriter& rewriter_;
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    CastCallBack(Rewriter& rewriter) : rewriter_(rewriter) {}
 
     void run(const MatchFinder::MatchResult &Result) override {
         // Your code goes here
+	const auto* CastExpr = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+	  if (CastExpr ->getCastKind() == CK_ToVoid)
+            return;
+	  if (CastExpr ->getExprLoc().isMacroID())
+            return;
+	  auto ReplaceRange = CharSourceRange::getCharRange(
+      		CastExpr->getLParenLoc(), CastExpr->getSubExprAsWritten()->getBeginLoc());
+	  StringRef text = 
+		    Lexer::getSourceText(CharSourceRange::getTokenRange(
+						CastExpr->getLParenLoc().getLocWithOffset(1), 
+                				CastExpr->getRParenLoc().getLocWithOffset(-1)),
+						*Result.SourceManager, Result.Context->getLangOpts());
+	  std::string replaceText = ("static_cast<" + text + ">").str();
+          auto castIgnore = CastExpr->getSubExprAsWritten()->IgnoreImpCasts();

```

Comment:
Please describe why you are using here `getSubExprAsWritten()` and `IgnoreImpCasts()`.
What would be the difference with the use of `getSubExprAsWritten()->IgnoreImpCasts()`, `getSubExpr()->IgnoreImpCasts()` and `getSubExprAsWritten()`?

Reply:
- getSubExprAsWritten() gets the cast subexpression as it was written in the source code, looking through any implicit casts or other intermediate nodes introduced by semantic analysis. Then we use IgnoreImpCasts() to ignore any implicit casts which might surround this expression.
- It doesn't matter if we used getSubExprAsWritten() alone or with IgnoreImpCasts(). getSubExprAsWritten()->IgnoreImpCasts(), getSubExpr()->IgnoreImpCasts() and getSubExprAsWritten() would all give the same result.

For example:
```
llvm::outs() << "getSubExprAsWritten()->IgnoreImpCasts(): " << castNode->getSubExprAsWritten()->IgnoreImpCasts()->getStmtClassName();
llvm::outs() << "getSubExpr()->ignoreImpCasts(): " << castNode->getSubExpr()->IgnoreImpCasts()->getStmtClassName();
llvm::outs() << "getSubExprAsWritten(): " << castNode->getSubExprAsWritten()->getStmtClassName();
```
Output will be:
```
getSubExprAsWritten()->IgnoreImpCasts(): DeclRefExpr
getSubExpr()->ignoreImpCasts(): DeclRefExpr
getSubExprAsWritten(): DeclRefExpr
```